app/controllers/partners/family_requests_controller.rb

```diff
@@ -13,6 +13,21 @@ def new
       @children = @filterrific.find
     end
 
+    def confirmation
+      @children_ids = params.keys.select { |key| key.start_with?('child-') && params[key] == 'true' }.map { |key| key.split('-').last.to_i }
+      children = current_partner.children.active.where(id: @children_ids).where.not(item_needed_diaperid: [nil, 0])
+
+      @items = {}
+      children.each do |child|
+        item = Item.find(child.item_needed_diaperid)
+        unless item.nil?
+          @items[item.name] = {first_name: child.first_name, last_name: child.last_name}
+        end
+      end
```

Comment:
This approach strikes me as pretty messy. Probably what I would do is just store the request details in the session. Basically store the full param hash in the session (once whatever validation has to happen is done). Then in the `create` action, you read it out of the session and delete it. This way you don't need to mess with passing hidden fields through the interstitial page. Only thing I'm unsure about is if the requests will be too big to fit in the session cookie.

Another option is to use Turbo Frames to render a popup before submitting the form. You send the parameters to the `confirmation` route, which renders some HTML in a partial (pretty much the same as you have now). Then when the dialog is submitted, you just submit the exact same parameters you already have in the form to the existing update method. Here's one approach to doing that: https://www.hotrails.dev/articles/rails-modals-with-hotwire

There's one more option - just dump the whole shebang in a JSON blob into a single hidden field.

Reply:
Ok I will try in the order you gave